src/main/java/org/folio/services/domainevent/CommonDomainEventPublisher.java

```diff
@@ -149,7 +149,7 @@ public <R> Future<Long> publishStream(ReadStream<R> readStream,
       });
 
     return promise.future()
-      .onComplete(e -> kafkaProducer.close())
+      .onComplete(e -> kafkaProducer.flush(x -> kafkaProducer.close()))
```

Comment:
This change is in production code, does that mean that failures could happen in production usage?

If so, what is the impact of those?

Reply:
Good question.

The original close has the impact of "close". The close returns a Future which is ignored. So it may not be closed when the method completes.

With the new code it flushes and closes, but also that is not guaranteed to be complete until completion which is why I think it's more proper to use eventually here.

A more fundamental question is whether close should be used at all. If the Kafka Client works like everything else in Vert.x, it's a pool and you better reuse it and not close at all. Using close+flush will only make things even slower.